crates/diffbot_lib/src/job/runner.rs

```diff
@@ -1,152 +1,52 @@
-use std::sync::Arc;
+use crate::github::{github_api::CheckRun, github_types::Output};
 
-use flume::Receiver;
-use tokio::sync::Mutex;
-
-use crate::{
-    github::github_types::CheckOutputs,
-    job::types::{Job, JobRunner},
-};
-
-use super::types::JobJournal;
-
-async fn handle_job<S: AsRef<str>, F>(name: S, job: Job, runner: F)
-where
-    F: JobRunner,
-{
-    println!(
-        "[{}] [{}#{}] [{}] Starting",
-        chrono::Utc::now().to_rfc3339(),
-        job.base.repo.full_name(),
-        job.pull_request,
-        job.check_run.id()
-    );
-    let _ = job.check_run.mark_started().await; // TODO: Put the failed marks in a queue to retry later
-                                                //let now = Instant::now();
-
-    let job_clone = job.clone();
-    let output = tokio::task::spawn_blocking(move || runner(&job_clone)).await;
-
-    println!(
-        "[{}] [{}#{}] [{}] Finished",
-        chrono::Utc::now().to_rfc3339(),
-        job.base.repo.full_name(),
-        job.pull_request,
-        job.check_run.id()
-    );
-
-    //eprintln!("Handling job took {}ms", now.elapsed().as_millis());
-
-    if let Err(e) = output {
-        let fuckup = match e.try_into_panic() {
-            Ok(panic) => match panic.downcast_ref::<&str>() {
-                Some(s) => s.to_string(),
-                None => "*crickets*".to_owned(),
-            },
-            Err(e) => e.to_string(),
-        };
-        eprintln!("Join Handle error: {}", fuckup);
-        let _ = job.check_run.mark_failed(&fuckup).await;
-        return;
-    }
-
-    let output = output.unwrap();
-    if let Err(e) = output {
-        let fuckup = format!("{:?}", e);
-        eprintln!("Other rendering error: {}", fuckup);
-        let _ = job.check_run.mark_failed(&fuckup).await;
-        return;
-    }
-
-    match output.unwrap() {
-        CheckOutputs::One(output) => {
-            let res = job.check_run.mark_succeeded(output).await;
-            if res.is_err() {
-                let _ = job
-                    .check_run
-                    .mark_failed(&format!("Failed to upload job output: {:?}", res))
-                    .await;
-            }
+pub async fn handle_output<S: AsRef<str>>(output: Vec<Output>, check_run: CheckRun, name: S) {
+    match output.len() {
+        0usize => {

```

Comment:
.. is this suffix actually necessary? I thought rust automatically converted number literals to the relevant type

Reply:
it's interpreted as u32 the last time i tried